lib/include/jxl/cms_interface.h

```diff
@@ -0,0 +1,210 @@
+/* Copyright (c) the JPEG XL Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** @file cms_interface.h
+ * @brief Interface to allow the injection of different color management systems
+ * (also called color management modules, or CMMs) in JPEG XL.
+ *
+ * A CMS is needed by the JPEG XL encoder and decoder to perform colorspace
+ * conversions. This defines an interface that can be implemented for different
+ * CMSes and then passed to the library.
+ */
+
+#ifndef JXL_CMS_INTERFACE_H_
+#define JXL_CMS_INTERFACE_H_
+
+#include "jxl/color_encoding.h"
+#include "jxl/memory_manager.h"
+
+#if defined(__cplusplus) || defined(c_plusplus)
+extern "C" {
+#endif
+
+/** Represents an input or output colorspace to a color transform, as a
+ * serialized ICC profile. */
+typedef struct {
+  /** The serialized ICC profile. */
+  struct {
+    const uint8_t* data;
+    size_t size;
+  } icc;
+
+  /** Structured representation of the colorspace, if applicable. */
+  JxlColorEncoding color_encoding;
+
+  /** Number of components per pixel. */
+  size_t num_channels;
```

Comment:
What are the possible values of this field and their meaning?

Reply:
For implementing all the color conversions needed to show a jxl image on a screen, we need at least the following:

- 1: Grayscale
- 3: RGB
- 4: CMYK

It might also be useful to support other things like L*a*b* or RGBA/RGBX, but it's probably best to keep this minimal so it's not too much work to implement this interface (3 cases is already enough).

Is the pixel format just implicitly floats with the components interleaved and a nominal range of [0..1] where 0 is black (i.e. no light or max ink)?

It is useful to spell out such conventions, especially for the CMYK case where [0..100] is also used, and 0 can either mean black (max ink) or white (no ink).

Reply:
I also meant that a comment for the user to know what to do with this field would be helpful

Reply:
For CMYK, 0=max ink is how it is done in jxl and in skcms, but not how it is done in lcms. See also https://github.com/libjxl/libjxl/pull/237

It's somewhat redundant if this is the same as the number of channels implied by the ICC profile, but it might be convenient to have it so implementations don't have to figure it out themselves, and it could be a good sanity check which might help to avoid buffer overrun bugs caused by someone thinking it's 3 components while in reality there is only 1 etc.

Reply:
Makes sense, thanks. Added, as well as a comment clarifying the CMYK representation (but the latter, I added below, to the documentation for `run`).